src/com/google/javascript/jscomp/CheckRequiresForConstructors.java

```diff
@@ -320,14 +320,32 @@ private void visitNewNode(NodeTraversal t, Node newNode) {
     }
   }
 
-  private void visitClassNode(Node classNode) {
+  private void visitClassNode(NodeTraversal t, Node classNode) {
     String name = NodeUtil.getName(classNode);
     if (name != null) {
       constructors.add(name);
     }
+    
     Node extendClass = classNode.getSecondChild();
+    
+    // If the superclass is something other than a qualified name, ignore it.
```

Comment:
nit: Make the code structure follow this comment with an early return:

if (!extendClass.isQualifiedName()) {
  return;
}

// Grab the root superclass namespace.
Node root = ...

In fact then we can probably drop the comment too.